### PR TITLE
fix: Clean up color system

### PR DESCRIPTION
**Pull request by misund**
This pull request:
- Moves the type color definitions from the colors objet to the skinColors object
- Removes the colors.button.* colors (uses colors.grayTint* instead)
- Uses getColor and getVariable in lots of more places

#### Please tick a box ###
- [x] **refactor** _(A code change that neither fixes a bug nor adds a feature_)

#### If you're adding a feature, is it documented in a storybook story?
- [x] Not adding a new feature

#### Are the components you're working on reusable between brands?
- [x] Not working on any components

#### If you're creating markup, did you add proper semantics? 
- [x] Not applicable

_(Did you do a CR and see that there is something that we should check for each PR, that are not on the list, please [update this document](https://github.com/dbmedialab/shiny/edit/master/.github/PULL_REQUEST_TEMPLATE.md))_

### CODE REVIEW

File: src/atoms/Button.js [hunk 1]
```diff
@@ -28,13 +28,15 @@ const Button = styled.button`
 	${props => (props.ALLCAPS && 'text-transform: uppercase')};
 
 	${props => props.outline && `
-		border: .1rem solid ${props.background? getColor(props.background):  props.theme.colors.button.background};
+		border: .1rem solid ${getColor(props.background)};
 		background: transparent;
-		color: ${props.background? getColor(props.background): props.theme.colors.button.background};
+		color: ${props.background};
```
Comment (tnlitv) [inline src/atoms/Button.js]: are you sure this check was unnecessary?

File: src/themes/mat/colors.js [hunk 2]
```diff
@@ -5,7 +5,7 @@ const colorsToShade= {
 	secondary: '#0096a3',
 	grey: '#7a7a7a',
 	darkGrey: '#4a4a4a',
-	type: '#333333',
+	darkness: '#333333',
```
Comment (evoX0) [inline src/themes/mat/colors.js]: what we are a win in this change?)

File: src/atoms/Button.js (hunk 1 above)
[the remark this answers is not in the document]
Comment (misund) [reply src/atoms/Button.js]: Thanks, should be getColor

File: src/themes/mat/colors.js (hunk 2 above)
Comment (misund) [reply src/themes/mat/colors.js]: (Mirroring [this change](https://github.com/dbmedialab/shiny/pull/565/commits/68a0b093b4c401c7d5fce0dc9aa37bfaa5fbfe7c#diff-461ba82a7711956629e733397c17c2a0R64) in the default theme)

**Pull request merged**